### PR TITLE
fix: [ANDROAPP-4023, ANDROAPP-4674] Search design review

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

**Review by ferdyrod — APPROVED** [review]

**Comment by andresmr** [comment]
I think UI test are failing: https://app-automate.browserstack.com/dashboard/v2/builds/6956a59f608843050e721fca123356259516a816

File: app/src/main/java/org/dhis2/Bindings/ViewExtensions.kt [hunk 1]
```diff
@@ -69,3 +74,43 @@ fun View.clipWithAllRoundedCorners(curvedRadio: Int = 16.dp) {
...
+            onItemSelected(selectedIndex)
+        }
+
+        override fun onNothingSelected(p0: AdapterView<*>?) {
+            // Don't do anything
+        }
+    }
+}
+
+fun FloatingActionButton.display(shouldBeDisplayed: Boolean) {
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/Bindings/ViewExtensions.kt:110]: not sure, but could this be replaced by the visible property of the floating action button?

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt [hunk 2]
```diff
@@ -55,6 +56,7 @@ class SearchTEIViewModel(
 
     private var searching: Boolean = false
     private var _filtersActive = MutableLiveData(false)
+    var filtersActive: LiveData<Boolean> = _filtersActive
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt:59]: Why it is a `var` if it is a `LiveData` object?

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt [hunk 3]
```diff
@@ -73,57 +75,68 @@ class SearchTEIViewModel(
...
+            searchForm = SearchForm(
+                queryHasData = queryData.isNotEmpty(),
+                minAttributesToSearch = searchRepository.getProgram(initialProgramUid)
+                    ?.minAttributesRequiredToSearch()
+                    ?: 1,
+                isForced = false,
+                isOpened = false
+            ),
+            searchFilters = SearchFilters(
+                hasActiveFilters = _filtersActive.value == true,
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt:137]: I think that `_filtersActive.value == true` is used four times in the class, do you think that could be refactored to a method like `private fun hasActiveFilters() = _filtersActive.value == true`?

**Review by JaimeToca — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**